File: src/main/java/net/reflact/engine/spells/Spell.java
package net.reflact.engine.spells;

import net.minestom.server.entity.Player;

public interface Spell {
    String getId();
    String getName();
    long getCooldownMillis();
    double getManaCost();

    /**
     * Called when the spell is successfully cast (cooldown and cost checks passed).
     */
    void onCast(Player caster);
}

File: src/main/java/net/reflact/engine/spells/Spell.kt
package net.reflact.engine.spells

import net.minestom.server.entity.Player

interface Spell {
    val id: String
    val name: String
    val cooldownMillis: Long
    val manaCost: Double

    /**
     * Called when the spell is successfully cast (cooldown and cost checks passed).
     */
    fun onCast(caster: Player)
}
